src/main/java/rx/internal/operators/OperatorSkipTimed.java

```diff
@@ -77,4 +72,15 @@ public void onCompleted() {
             }
         };
     }
+    
+    final static class AtomicBooleanAction extends AtomicBoolean implements Action0 {
```

Comment:
Yes, this is better. You don't use CAS on the boolean thus a regular  volatile field suffices. Then you have 1 allocation. Also switching to OnSubscribe saves another.